edge/edge.go

```diff
@@ -217,11 +236,54 @@ func (manager *Manager) checkDockerRuntimeConfig() error {
 			return err
 		}
 
-		err = manager.stackManager.stop()
+		err = manager.stackManager.Stop()
 		if err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
+
+func GetNewHttpClient(timeout float64, options *agent.Options) *http.Client {
+	httpCli := &http.Client{
+		Timeout: time.Duration(timeout) * time.Second,
+	}
+
+	if options.EdgeInsecurePoll {
```

Comment:
move all below to an internal function
```suggestion
	httpCli.Transport = buildTransport(options)
	return httpCli
	....
	....
	func buildTransport(options *agent.Options) *http.Transport {
	     if options.SSLCert != "" && options.SSLKey != "" {
	         // return SSL Transport
	     }
	     
	     if options.EdgeInsecurePoll {
	        // return InsecureSkipVerify Transport
	     }
	     
	     return nil
	 }
```